wdl/ExpansionHunterScatter.wdl

```diff
@@ -42,18 +54,90 @@ workflow ExpansionHunterScatter {
                 bam_or_cram_index=bam_or_cram_index_,
                 reference_fasta=reference_fasta,
                 reference_fasta_index=reference_fasta_index_,
-                variant_catalog=variant_catalog,
+                split_variant_catalogs=svc.catalogs_json,
                 sample_id=sample_id,
                 ped_file=ped_file,
                 expansion_hunter_docker=expansion_hunter_docker,
-                runtime_attr=runtime_attr
+                python_docker=python_docker
         }
     }
 
     output {
         Array[File] jsons = expanionHunter.json
-        Array[File] vcfs = expanionHunter.vcf
+        Array[File] vcfs_gz = expanionHunter.vcf_gz
         Array[File] overlapping_reads = expanionHunter.overlapping_reads
         Array[File] timing = expanionHunter.timing
     }
 }
+
+task SplitVariantCatalog {
+    input {
+        File variant_catalog
+        Int batch_size
+        String output_prefix
+        String python_docker
+        RuntimeAttr? runtime_attr_override
+    }
+
+    output {
+        Array[File] catalogs_json = glob("~{output_prefix}*.json")
+    }
+
+    command <<<
+        set -euxo pipefail
+
+        python <<CODE
+        import json
+        from pathlib import Path
+
+        filename = "~{variant_catalog}"
+        filename_without_ext = Path(filename).stem
+        output_prefix = "~{output_prefix}"
+        i = 0
+        subset_counter = 0
+
+
+        def serialize():
+            with open(f"{output_prefix}{filename_without_ext}_{subset_counter:06}.json", "w") as f_out:
+                json.dump(subset_catalogs, f_out, indent=4)
+
+
+        with open(filename, "r") as f_in:
+            catalogs = json.load(f_in)
+
+            subset_catalogs = []
+            for catalog in catalogs:
+                subset_catalogs.append(catalog)
+                i += 1
+                if i == ~{batch_size}:
+                    serialize()
+                    i = 0
+                    subset_counter += 1
+                    subset_catalogs = []
+            if len(subset_catalogs) > 0:
+                serialize()
+        CODE
+    >>>
+
+    RuntimeAttr runtime_attr_str_profile_default = object {
+        cpu_cores: 1,
+        mem_gb: 4,
+        boot_disk_gb: 10,
+        preemptible_tries: 3,
+        max_retries: 1,
+        disk_gb: 10
+    }
+    RuntimeAttr runtime_attr = select_first([
+        runtime_attr_override,
+        runtime_attr_str_profile_default])
+
+    runtime {
+        docker: python_docker
+        cpu: runtime_attr.cpu_cores
+        memory: runtime_attr.mem_gb + " GiB"
+        disks: "local-disk " + runtime_attr.disk_gb + " HDD"
+        bootDiskSizeGb: runtime_attr.boot_disk_gb
+        preemptible: runtime_attr.preemptible_tries
+        maxRetries: runtime_attr.max_retries
+    }
```

Comment:
Didn't notice this before, but this looks like a simplified version of what we usually do. The `RuntimeAttr` struct looks like this:
```
struct RuntimeAttr {
    Float? mem_gb
    Int? cpu_cores
    Int? disk_gb
    Int? boot_disk_gb
    Int? preemptible_tries
    Int? max_retries
}
```
So each member of the struct is Optional. This is because the user is meant to be able to provide a subset of the members, for example to only change the memory they would provide:
```
"runtime_attr_my_task": { "mem_gb": "8" },
```
The way it's written here, the workflow would crash because it would attempt to use the undefined fields (`runtime_attr.cpu_cores`, `runtime_attr.disk_gb`, etc.). This is why we use a more complicated looking runtime section like this for most tasks:
```
  runtime {
    cpu: select_first([runtime_attr.cpu_cores, default_attr.cpu_cores])
    memory: mem_gb + " GiB"
    disks: "local-disk " + select_first([runtime_attr.disk_gb, default_attr.disk_gb]) + " HDD"
    bootDiskSizeGb: select_first([runtime_attr.boot_disk_gb, default_attr.boot_disk_gb])
    docker: sv_pipeline_docker
    preemptible: select_first([runtime_attr.preemptible_tries, default_attr.preemptible_tries])
    maxRetries: select_first([runtime_attr.max_retries, default_attr.max_retries])
  }
```
Here it will properly select overrides if they are provided, but use defaults if they are missing.

Reply:
Agree, that will make it more user-friendly, thanks. I updated accordingly.